tests/suites/test_suite_x509parse.function

```diff
@@ -1250,6 +1250,86 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_FS_IO:MBEDTLS_X509_CRT_PARSE_C:MBEDTLS_CERTS_C:MBEDTLS_SHA1_C:MBEDTLS_RSA_C */
+void mbedtls_x509_crt_parse_subjectkeyid( char * crt_path, char *subjectKeyId, int subjectKeyIdLength, int retVal )
+{
+    mbedtls_x509_crt crt;
+    int i = 0;
+    int result = 0;
+
+    mbedtls_x509_crt_init( &crt );
+
+    TEST_ASSERT( mbedtls_x509_crt_parse_file( &crt, crt_path ) == retVal );
+
+    if(retVal != 0)
+    {
+        while(i < subjectKeyIdLength)
+        {
+            result |= crt.subject_key_id.p[i] != subjectKeyId[i*2];
+            result |= crt.subject_key_id.p[i+1] != subjectKeyId[i*2+1];
+            i++;
+        }
+
+        TEST_ASSERT(result == 0);
```

Comment:
THe original code was erronous so it has been removed, but in some cases I could use ASSERT_COMPARE.
Somehow if I want to pass byte array like this "\x61\x62" etc. It is not working in the tests. The first item is not "a", but "\"...And the keyID-s can contain many special non ASCII characters, so I cannot use these directly between apostrophes. I removed checking the values of this elements, only the length, tag and success of parsing is checked now. I haven't found anyone checking for example the SerialNumber in the tests (this would be similar as my problem), so I hope this is good enough

Reply:
I'm not sure I understand the problem here, but for most test functions that take a byte string as input, we use the type `data_t *`, and the `.data` file contains the hexadecimal representation of the data. The string parser for `.data` files does not support C escape sequences like `\xNN`.

Reply:
Yes, that is my problem. Lets say that subject key identifier equals: 0xC0  (only 1 byte long)
I cannot provide it via argument as hex (\xC0), and it the parser also throws error if I try to pass it as a string ("À"). So I just removed the checking of the concrete value, I only assert the length and the tag of it.

Do you have any idea how to test this kind of values? I haven't found a solution for this (e.g with serial number, that is a same type of data). Thank you!